### PR TITLE
feat: Add `StandardScaler` transformer

### LINKED ISSUE #142: Standardize `Table`
### Is your feature request related to a problem?

Standardization means to give a variable a mean of 0 and a standard deviation of 1.

### Desired solution

* Add a class `StandardScaler` in `safeds.data.tabular.transformation` that wraps the [`StandardScaler` of `scikit-learn`](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.StandardScaler.html)
* Make it a subclass of `InvertibleTableTransformer`

### Considered alternatives

* Call the class `Standardizer`. However, several strategies (e.g. #141) for features _scaling_ exist and the suffix `Scaler` makes them more visible.

### PR DESCRIPTION
**Pull request by zzril**
Closes #142.

### Summary of Changes

* Added new class `StandardScaler` in `tabular/transformation`.
* Added tests.
* Added helper method `check_that_tables_are_close`.

Co-authored-by: sibre28 <86068340+sibre28@users.noreply.github.com>

### CODE REVIEW

**Comment by lars-reimann** [comment]
## [🦙 MegaLinter](https://megalinter.io/6.22.2) status: ❌ [ERROR](https://github.com/Safe-DS/Stdlib/actions/runs/5025221073)

| Descriptor  |                                 Linter                                 |Files|Fixed|                            Errors                            |Elapsed time|
|-------------|------------------------------------------------------------------------|-----|----:|--------------------------------------------------------------|------------|
|✅ PYTHON    |[black](https://megalinter.io/6.22.2/descriptors/python_black)          |    3|    1|                                                             0|0.66s       |
|❌ PYTHON    |[mypy](https://megalinter.io/6.22.2/descriptors/python_mypy)            |    3|     |[1](https://github.com/Safe-DS/Stdlib/actions/runs/5025221073)|1.56s       |
|✅ PYTHON    |[ruff](https://megalinter.io/6.22.2/descriptors/python_ruff)            |    3|    1|                                                             0|0.02s       |
|✅ REPOSITORY|[git_diff](https://megalinter.io/6.22.2/descriptors/repository_git_diff)|yes  |     |no                                                            |0.01s       |

See detailed report in [MegaLinter reports](https://github.com/Safe-DS/Stdlib/actions/runs/5025221073)
_Set `VALIDATE_ALL_CODEBASE: true` in mega-linter.yml to validate all sources, not only the diff_

_MegaLinter is graciously provided by [OX Security](https://www.ox.security/?ref=megalinter)_

File: tests/safeds/data/tabular/transformation/test_standard_scaler.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+        )
+        transformer = transformer.fit(table, None)
+        with pytest.warns(
+            UserWarning,
+            match="StandardScaler only changes data within columns, but does not remove any columns.",
+        ):
+            assert transformer.get_names_of_removed_columns() == []
+
+
+class TestInverseTransform:
```
Comment (jxnior01) [inline tests/safeds/data/tabular/transformation/test_standard_scaler.py:196]: Nice, you could as well add a test method that should handle edge cases such as handling of passed empty tables or empty columns.

File: src/safeds/data/tabular/transformation/_standard_scaler.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        -------
+        added_columns : list[str]
+            A list of names of the added columns, ordered as they will appear in the table.
+
+        Raises
+        ------
+        TransformerNotFittedError
+            If the transformer has not been fitted yet.
+        """
+        warnings.warn("StandardScaler only changes data within columns, but does not add any columns.", stacklevel=1)
```
Comment (lars-reimann) [inline src/safeds/data/tabular/transformation/_standard_scaler.py:143]: I don't think these warnings are useful. Just returning an empty list already expresses that.

File: src/safeds/data/tabular/transformation/_standard_scaler.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+        -------
+        removed_columns : list[str]
+            A list of names of the removed columns, ordered as they appear in the table the StandardScaler was fitted on.
+
+        Raises
+        ------
+        TransformerNotFittedError
+            If the transformer has not been fitted yet.
+        """
+        warnings.warn("StandardScaler only changes data within columns, but does not remove any columns.", stacklevel=1)
```
Comment (lars-reimann) [inline src/safeds/data/tabular/transformation/_standard_scaler.py:181]: See above.

File: src/safeds/data/tabular/transformation/_standard_scaler.py (hunk 2 above)
Comment (zzril) [reply src/safeds/data/tabular/transformation/_standard_scaler.py]: Removed those warnings (and corresponding tests).  
We also have these warnings currently in the `RangeScaler`, will create a new issue to remove them.

File: src/safeds/data/tabular/transformation/_standard_scaler.py (hunk 3 above)
Comment (zzril) [reply src/safeds/data/tabular/transformation/_standard_scaler.py]: Fixed.

File: src/safeds/data/tabular/transformation/_standard_scaler.py (hunk 2 above)
Comment (zzril) [reply src/safeds/data/tabular/transformation/_standard_scaler.py]: Created issue (#323) and already submitted PR for it (#324).

File: tests/helpers/_resources.py [hunk 4]
```diff
@@ -18,3 +22,25 @@ def resolve_resource_path(resource_path: str | Path) -> str:
...
+    """
+    assert table1.schema == table2.schema
+    for column_name in table1.column_names:
+        assert table1.get_column(column_name).type == table2.get_column(column_name).type
+        assert table1.get_column(column_name).type.is_numeric()
+        assert table2.get_column(column_name).type.is_numeric()
+        for i in range(table1.number_of_rows):
+            entry_1 = table1.get_column(column_name).get_value(i)
+            entry_2 = table2.get_column(column_name).get_value(i)
+            assert (isclose(entry_1, entry_2))
```
Comment (lars-reimann) [inline tests/helpers/_resources.py:46]: The `approx` method of `pytest` should yield a better error message if the assertion fails:

https://stackoverflow.com/a/39623614

**Comment by lars-reimann** [comment]
@zzril 2a670cb69ab520e7f08ba1772d3302f6545299dd fixed the mypy error for whatever reason.

**Comment by zzril** [comment]
> @zzril [2a670cb](https://github.com/Safe-DS/Stdlib/commit/2a670cb69ab520e7f08ba1772d3302f6545299dd) fixed the mypy error. It seems that mypy cannot handle something in `tests/helpers/_resources.py`. Because the file is no longer changee now, it is also no longer linted, so the error went away.

I think the issue was that the entry of the new helper method in `tests`/`helpers`/`__init.py__` was missing.
